remotes/docker/scope.go

```diff
@@ -51,21 +51,31 @@ func contextWithRepositoryScope(ctx context.Context, refspec reference.Spec, pus
 	if err != nil {
 		return nil, err
 	}
-	return context.WithValue(ctx, tokenScopesKey{}, []string{s}), nil
+	return contextWithScope(ctx, s), nil
 }
 
-// contextWithAppendPullRepositoryScope is used to append repository pull
-// scope into existing scopes indexed by the tokenScopesKey{}.
-func contextWithAppendPullRepositoryScope(ctx context.Context, repo string) context.Context {
+func contextWithScope(ctx context.Context, scope string) context.Context {
```

Comment:
I don't think having a private version is necessary here

Reply:
fixed